GitVersionCore/OutputVariables/VariableProvider.cs

```diff
@@ -1,71 +1,47 @@
+        public static VersionVariables GetVariablesFor(SemanticVersion semanticVersion, AssemblyVersioningScheme assemblyVersioningScheme, VersioningMode mode)
         {
             var bmd = semanticVersion.BuildMetaData;
-            var formatter = bmd.Branch == "develop" ? new CiFeedFormatter() : null;
-            
-            var variables = new Dictionary<string, string>(StringComparer.InvariantCultureIgnoreCase)
+            IFormatProvider formatProvider = null;
+            if (mode == VersioningMode.ContinuousDeployment)
             {
-                {Major, semanticVersion.Major.ToString()},
-                {Minor, semanticVersion.Minor.ToString()},
-                {Patch, semanticVersion.Patch.ToString()},
-                {PreReleaseTag, semanticVersion.PreReleaseTag},
-                {PreReleaseTagWithDash, semanticVersion.PreReleaseTag.HasTag() ? "-" + semanticVersion.PreReleaseTag : null},
-                {BuildMetaData, bmd},
-                {FullBuildMetaData, bmd.ToString("f")},
-                {MajorMinorPatch, string.Format("{0}.{1}.{2}", semanticVersion.Major, semanticVersion.Minor, semanticVersion.Patch)},
-                {SemVer, semanticVersion.ToString(null, formatter)},
-                {LegacySemVer, semanticVersion.ToString("l", formatter)},
-                {LegacySemVerPadded, semanticVersion.ToString("lp", formatter)},
-                {AssemblySemVer, semanticVersion.GetAssemblyVersion(assemblyVersioningScheme)},
-                {AssemblyFileSemVer, semanticVersion.GetAssemblyFileVersion(assemblyVersioningScheme)},
-                {FullSemVer, semanticVersion.ToString("f", formatter)},
-                {InformationalVersion, semanticVersion.ToString("i", formatter)},
-                {ClassicVersion, string.Format("{0}.{1}", semanticVersion.ToString("j"), (bmd.CommitsSinceTag ?? 0))},
-                {ClassicVersionWithTag, string.Format("{0}.{1}{2}", semanticVersion.ToString("j"),
-                    bmd.CommitsSinceTag ?? 0,
-                    semanticVersion.PreReleaseTag.HasTag() ? "-" + semanticVersion.PreReleaseTag : null)},
-                {BranchName, bmd.Branch},
-                {Sha, bmd.Sha},
-            };
+                // For continuous deployment the commits since tag gets promoted to the pre-release number
+                semanticVersion = new SemanticVersion(semanticVersion);
+                if (semanticVersion.PreReleaseTag.HasTag())
+                {
+                    var oldPreReleaseNumber = semanticVersion.PreReleaseTag.Number;
+                    semanticVersion.PreReleaseTag.Number = semanticVersion.BuildMetaData.CommitsSinceTag;
+                    semanticVersion.BuildMetaData.CommitsSinceTag = oldPreReleaseNumber;
+                }
+                else
+                {
+                    // When there is no pre-release tag we will make the version a 4 part number
+                    formatProvider = new CommitsAsFourthVersionPartFormatter();
```

Comment:
I would only do this if `semanticVersion.BuildMetaData.CommitsSinceTag` is not 0. Version like `1.2.3.0` seems wrong to me.